docs/Guides/creating_new_vendor_app.md

```diff
@@ -0,0 +1,137 @@
+
+![image.png](../../assets/images/guides/vendor_application/authorization_request_resolver.png)
+
+
+
+Let's create a class variable called defaultRequestResolver of type Oauth2AuthorizationRequestResolver to get the default configurations and during the constructor initialization, let's initialize this variable with the clientRegistrationRepo for vendasta and the auth url which was hardcoded from the previous step.
+
+The class should implement the Oauth2AuthorizationRequestResolver so that means we need to implement the resolve methods. But since we are only interested in integrating Vendasta as the provider let's use the same logic for both the overridden overloaded methods(That’s a mouthful huh!! Welcome to java) . 
+
+Let's create a method `createCustomEntry` to take care of this activity.
+
+![image.png](../../assets/images/guides/vendor_application/create_custom_entry.png)
+
+
+The method does a lot of things and it could be engineered better, but to make it screenshot friendly, we have made everything available here. Let's look at what’s happening here
+
+1. Let's call the resolve method of the default implementation of this class by spring boot . (We need this because we need various configuration setup like adding the clientRegistrationId,state,redirectUrl and such. lets create a temp Oauth2AuthorizationRequest.
+
+2. The second line  gets the accountId from the request Url(This is a mandatory step when integrating with Vendasta. When creating the entry Url , The accountId should be tagged in the format `/entry/accountId` for it to work properly. This is explained better in the docs above.
+
+3. The next if statement will make sure that we redirect the user to the right location after the login. This also works hand in hand with another problem. That is to look for an active session with the client. If there is no active session with the client then this variable is null. So the auth flow happens only when there is no valid sessions. 
+
+4. The next line sets the accountId in the `state` parameter of your authUrl.
+
+5. We then set the callback url here using a session variable called `baseUrl` ( This is purely optional. An authenticated session can be validated a number of ways. We are using a simple session variable. On successful validation from Vendasta we get a request to the redirect Url . Since we don’t want to send the user to the redirect url always we use this variable. Let's look at this implementation later
+
+6. if the baseUrl already has a valid session then we just enter into the else part and make the object authorizationRequestResolver as `null` because we don't want to trigger an auth flow once authorized.
+
+7. The next few lines are self explanatory. They add the accountId to the `authUrl`. Our final Url would look something like this.
```

Comment:
Saying something is self explanatory, and then adding context is redundant. We can remove the self explanatory statement... that probably isn't something that should appear in technical documentation.